### PR TITLE
[bitnami/contour] feat: support existing secret for tls

### LINKED ISSUE #3766: [bitnami/contour] Use certs in secret rather that using `contour cert` job
**Which chart**:
The `contour` chart, version `2.1.1`

**Is your feature request related to a problem? Please describe.**
The `contour cert` job doesn't allow for automatic certificate renewal. Allowing to use custom secrets would allow the chart users to use tools like [cert-manager](https://cert-manager.io/) to handle the `mTLS` between `contour` and `envoy`

**Describe the solution you'd like**
Add a field in `values.yml` allowing the chart users to use a secret instead of the `contour cert` job for the `contour` and `envoy` TLS credentials.

**Describe alternatives you've considered**
An alternative would be to mount the secrets as additional volumes and find a way to tell `contour` and `envoy` to use them instead.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by dani8art**
**Description of the change**

- Added `tlsExistingSecret`, `contour.tlsExistingSecret` and `envoy.tlsExistingSecret` in order to support using custom existing secrets for tls.

- Fixed some linter errors.

**Benefits**

We will be able to use custom certificates.

**Possible drawbacks**

N/A

**Applicable issues**

  - fixes #3766

**Additional information**

Test script

- Please use the first commit to review the changes [44d096b](https://github.com/bitnami/charts/commit/44d096b6ca4900552c33e0a0014c88a5e818a41a), the second one is just linter issues.

```bash
helm template contour bitnami/contour/ -s templates/certgen/job.yaml --set "${HELM_TESTS_EXTRA_FLAGS}"
helm template contour bitnami/contour/ -s templates/certgen/rbac.yaml --set "${HELM_TESTS_EXTRA_FLAGS}"
helm template contour bitnami/contour/ -s templates/certgen/serviceaccount.yaml --set "${HELM_TESTS_EXTRA_FLAGS}"
helm template contour bitnami/contour/ -s templates/contour/deployment.yaml --set "${HELM_TESTS_EXTRA_FLAGS}" | grep secretName
helm template contour bitnami/contour/ -s templates/envoy/daemonset.yaml --set "${HELM_TESTS_EXTRA_FLAGS}" | grep secretName
```

Test cases

```bash
## Test Case: default config with certgen
export HELM_TESTS_EXTRA_FLAGS=""
## Test Case: use global tls existing secret
export HELM_TESTS_EXTRA_FLAGS="tlsExistingSecret=my-global-existing-secret"
## Test Case: use global tls existing secret and override using contour specific
export HELM_TESTS_EXTRA_FLAGS="tlsExistingSecret=my-global-existing-secret,contour.tlsExistingSecret=my-contour-existing-secret"
## Test Case: use global tls existing secret and override using envoy specific
export HELM_TESTS_EXTRA_FLAGS="tlsExistingSecret=my-global-existing-secret,envoy.tlsExistingSecret=my-envoy-existing-secret"
## Test Case: use envoy specific
export HELM_TESTS_EXTRA_FLAGS="envoy.tlsExistingSecret=my-envoy-existing-secret"
## Test Case: use contour and envoy specific
export HELM_TESTS_EXTRA_FLAGS="contour.tlsExistingSecret=my-contour-existing-secret,envoy.tlsExistingSecret=my-envoy-existing-secret"
```

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

**Comment by GrandChaman** [comment]
Wow, thank you very much for the **fast** PR ! Looks great to me :)

File: bitnami/contour/values.yaml [hunk 1]
```diff
@@ -44,18 +44,18 @@ configInline:
   # disable ingressroute permitInsecure field
   disablePermitInsecure: false
   tls:
-  #   minimum TLS version that Contour will negotiate
-  #   minimum-protocol-version: "1.1"
-  # Defines the Kubernetes name/namespace matching a secret to use
-  # as the fallback certificate when requests which don't match the
-  # SNI defined for a vhost.
+    #   minimum TLS version that Contour will negotiate
+    #   minimum-protocol-version: "1.1"
```
Comment (marcosbc) [inline bitnami/contour/values.yaml:48]: In cases like these (and throughout the rest of the files), I think we should be removing the spaces before the object, i.e.:

```suggestion
    # minimum TLS version that Contour will negotiate
    # minimum-protocol-version: "1.1"
```

Please make sure to review all occurrences like these, as I've seen many.

File: bitnami/contour/templates/_helpers.tpl [hunk 2]
```diff
@@ -33,6 +33,33 @@ Create the name of the contour-certgen service account to use
...
+{{- define "contour.contour.certs-secret.name" -}}
+{{- $existingSecret := default .Values.tlsExistingSecret .Values.contour.tlsExistingSecret -}}
+{{- $name := default "contourcert" $existingSecret -}}
+{{- printf "%s" $name -}}
+{{- end -}}
+
+{{/*
+Envoy certs secret name
+*/}}
+{{- define "contour.envoy.certs-secret.name" -}}
```
Comment (marcosbc) [inline bitnami/contour/templates/_helpers.tpl:57]: Is this being used anywhere? I cannot find any match.

Comment (dani8art) [reply bitnami/contour/templates/_helpers.tpl:57]: `bitnami/contour/templates/envoy/daemonset.yaml`

File: bitnami/contour/values.yaml (hunk 1 above)
Comment (dani8art) [reply bitnami/contour/values.yaml]: done!

File: bitnami/contour/values-production.yaml [hunk 3]
```diff
@@ -45,19 +45,19 @@ configInline:
...
-  # as the fallback certificate when requests which don't match the
-  # SNI defined for a vhost.
+    # minimum TLS version that Contour will negotiate
+    # minimum-protocol-version: "1.1"
+    # Defines the Kubernetes name/namespace matching a secret to use
+    # as the fallback certificate when requests which don't match the
+    # SNI defined for a vhost.
     fallback-certificate:
   #   name: fallback-secret-name
   #   namespace: '{{ .Release.Namespace }}'
```
Comment (marcosbc) [inline bitnami/contour/values-production.yaml:55]: I think this is still wrong

**Review by marcosbc — APPROVED** [review]
LGTM, please have a look at my comment but apart from that it looks great.

File: bitnami/contour/values-production.yaml (hunk 3 above)
Comment (dani8art) [reply bitnami/contour/values-production.yaml:55]: If I put it in the other way, the linter is failing.

**Pull request merged**